libres/tests/analysis/ies/test_ies_enkf_main.cpp

```diff
@@ -38,51 +35,142 @@ TEST_CASE("ies_enkf_linalg_extract_active", "[analysis]") {
     matrix_iset(Ein, 2, 1, 3.5);
 
     // minimal config needed to set initial data of `iens_enkf_data_type`
-    ies_enkf_config_set_ies_debug(ies_enkf_data_get_config(data), false);
     ies_enkf_config_set_ies_logfile(ies_enkf_data_get_config(data),
                                     "log_test_ies_enkf_linalg_extract_active");
     FILE *test_log = ies_enkf_data_open_log(data);
     ies_enkf_data_store_initialE(data, Ein);
     ies_enkf_data_fclose_log(data);
 
     // Test that `ies_enkf_linalg_extract_active` does nothing when all observations and realizations are active
-    matrix_type *E = matrix_alloc(state_size, ens_size);
-    ies_enkf_linalg_extract_active(data, E, stdout, false);
-    REQUIRE(matrix_equal(Ein, E));
+    {
```

Comment:
```suggestion
Section("ies_enkf_linalg_extract_active does nothing when all are active"){
```

Reply:
I would like to keep this test implementation as it is. The reason is that the `ies_enkf_data` structure maintains state throughout the entire test, and by introducing sections now for this particular test I fear there will be changes in behavior due to accumulated state. One of the main goals of the ongoing ies refactoring (not yet in this PR) is to reduce this state, and manage it a more transparent way.

Reply:
From my understanding of how `catch2` works, it is safe to use `SECTION` in this case, as the test-framework allows us to clean-up before each section is run.
Why is this different?

Reply:
The situation is opposite - the test holdes on to state which is mutated systematically through the test. 

It is certainly possible to rewrite to use sections - I suffered from a little review fatigue at the time.

Reply:
I had a check and the required change in the test is minimal: 

```cpp
    SECTION("ies_enkf_linalg_extract_active does nothing when all observations and realizations are active"){
        matrix_type *E = ies_enkf_alloc_activeE(data);
        REQUIRE(matrix_equal(Ein, E));
        matrix_free(E);
    }

    SECTION("ies_enkf_linalg_extract_active can deactivate an ensemble") {
        bool_vector_iset(ens_mask, 1, false);
        ies_enkf_data_update_ens_mask(data, ens_mask);
        matrix_type *E = ies_enkf_alloc_activeE(data);
        REQUIRE(matrix_iget(E, 0, 0) == 1.0);
        REQUIRE(matrix_iget(E, 1, 0) == 2.0);
        REQUIRE(matrix_iget(E, 2, 0) == 3.0);
        REQUIRE(matrix_get_rows(E) == 3);
        REQUIRE(matrix_get_columns(E) == 1);
        matrix_free(E);
    }

    SECTION("ies_enkf_linalg_extract_active can deactivate an observation") {
        bool_vector_iset(obs_mask, 1, false);
        ies_enkf_update_obs_mask(data, obs_mask);
        matrix_type *E = ies_enkf_alloc_activeE(data);
        REQUIRE(matrix_iget(E, 0, 0) == 1.0);
        REQUIRE(matrix_iget(E, 1, 0) == 3.0);
        REQUIRE(matrix_get_rows(E) == 2);
        REQUIRE(matrix_get_columns(E) == 2);
        matrix_free(E);
    }
```

Reply:
> the last assert needs to be changed to REQUIRE(matrix_get_columns(E) == 2);:

That is because of the state carrying I have argued about. Anyway - I'll update.

Reply:
Done